lib/epochs/index.js

```diff
@@ -530,46 +432,150 @@ function getGroupInit(ssb, groupId, cb) {
-      preferredEpoch = tips[0]
-    } else if (
-      isSubsetOf(members2, members1)
-    ) {
-      preferredEpoch = tips[1]
-    }
-    // case 4.6 - groups have overlapping membership, but disjoint
-    // case 4.7 - disjoint membership (no overlap!)
+  return epochs.find((epoch) => epoch.secret.equals(winningKey))
+}
+
+function BuildPreferredEpoch(ssb, groupId) {
+  return function buildPreferredEpoch(reduce, cb) {
+    const tips = getTipsFromTangleReduce(reduce)
+
+    var preferredEpoch
+    if (tips.length === 1) preferredEpoch = tips[0]
+    else if (tips.length === 2) {
+      const [members0, members1] = tips.map((t) => t.members)
+
+      // case 4.4 - same membership
+      if (isSameSet(members0, members1)) {
+        preferredEpoch = tieBreak(tips)
+      }
+
+      // case 4.5 - one group has a membership that's a subset of other
+      else if (isSubsetOf(members0, members1)) {
+        preferredEpoch = tips[0]
+      } else if (isSubsetOf(members1, members0)) {
+        preferredEpoch = tips[1]
+      }
+
+      // case 4.6 - groups have overlapping membership, but disjoint
+      else if (
+        intersection(members0, members1).size > 0
+      ) {
+        // choose one for now,
+        preferredEpoch = tieBreak(tips)
+        // but also kick off resolution
+        fixDisjointEpochsLater(ssb, groupId) // <<< DELAYED SIDE EFFECTS!
+      }
+
+      // case 4.7 - disjoint membership (no overlap!)
 
+      // prettier-ignore
+      else return cb(Error('Membership case not handled yet'))
+    } else return cb(Error(`case of ${tips.length} tips not handled yet`))
+
+    if (preferredEpoch.members) delete preferredEpoch.members
+
+    cb(null, preferredEpoch)
+  }
+}
+
+// similar to getTipEpochs but a little slower because loading the membership of the epochs
+// NOTE could optimise to just load the membership of ONLY the tips later :shrug:
+function getTipEpochsWithMembers(ssb, groupId, cb) {
+  if (cb === undefined)
+    return p(getTipEpochsWithMembers).call(this, ssb, groupId)
+
+  const opts = { getters: Getters(ssb) }
+  epochsReduce(ssb, groupId, opts, (err, reduce) => {
     // prettier-ignore
-    else return cb(Error('Membership case not handled yet'))
-  } else return cb(Error(`case of ${tips.length} tips not handled yet`))
+    if (err) return cb(clarify(err, 'Failed to resolve epoch @tangle/reduce'))
+
+    const tips = getTipsFromTangleReduce(reduce)
+    return cb(null, tips)
+  })
+}
+
+function fixDisjointEpochsLater(ssb, groupId) {
+  const timeout =
+    ssb?.config?.tribes2?.disjointResloveDelay ??
+    (110 * Math.random() + 10) * 1000 // 10-120 seconds
+  const timeoutId = setTimeout(() => {
+    // check if disjoint still
+    getTipEpochsWithMembers(ssb, groupId, (err, tips) => {
+      if (err)
+        return console.error(
+          clarify(err, 'Failed to check disjoint but overlapping state')
+        )
 
-  delete preferredEpoch.members
+      if (!isDisjointOverlappingState(tips)) return
```

Comment:
don't you already check this on line 469?

Reply:
aah this is a check for after the timeout. but still, can't you use the same logic/function both here and on 469?

Reply:
Can't use the logic on line 469 as that is insufficient - note it's part way into an if,else if, else if, so other things have been ruled out earlier.

Keeping this as is for the moment